tests/python/test_relational_vcfs.py

```diff
@@ -0,0 +1,90 @@
+import nose
+
+import tempfile
+
+
+
+def test_vcf_to_csv():
+    import vcf, csv
+    from common.relational_vcf import vcf_to_csv
+    vcfreader = vcf.Reader(open('tests/data/snv.vcf'))
+    columns = ['contig', 'position', 'sample:DP', 'info:VAF', 'info:DP', 'something']
+    filename = vcf_to_csv(vcfreader, columns, None,
+                          default_values={'something': 'Something!'})
+
+    with open(filename) as fd:
+        rows = list(csv.reader(fd))
+
+    # first two are from the same sample
+    assert rows[0] == ['20', '61795', '44', '', '81', 'Something!']
+    assert rows[1] == ['20', '61795', '37', '', '81', 'Something!']
+    # this is the last sample
+    assert rows[-1] == ['20', '75254', '40', '', '74', 'Something!']
+    assert len(rows) == 20 # 10 records, 2 samples each = 20 rows
+
+
+def test_csv_to_vcf():
+    import vcf, csv
+    from common.relational_vcf import vcf_to_csv, genotypes_to_records
+
+    # Here we just write the vcf to a CSV like we did above.
+    vcfreader = vcf.Reader(open('tests/data/one-sample.vcf'))
+    columns = ['contig', 'position', 'id', 'reference', 'alternates',
+               'quality', 'filters', 'info:DP', 'info:SS',
+               'info:SSC', 'info:GPV', 'info:SPV', 'info:SOMATIC',
+               'sample:GT','sample:GQ','sample:DP','sample:RD',
+               'sample:AD','sample:FREQ','sample:DP4',
+               'sample_name']
+    filename = vcf_to_csv(vcfreader, columns, None)
+
+    # Now we convert the rows in this CSV into dict (converting the position as
+    # well), which is what it would look like select from Postgres.
+    with open(filename) as fd:
+        rows = list(csv.reader(fd))
+        relations = []
+        for row in rows:
+            relation = {
+                col: val if val != '' else None
+                for col, val in zip(columns, row)
+            }
+            relation['position'] = int(relation['position'])
+            relations.append(relation)
+
+    text = open('tests/data/one-sample.vcf').readlines()
+    header = (l for l in text if l.startswith('#'))
+    template = vcf.Reader(header)
+
+    # Here we convert those dicts to vcf.Records, for later writing.
+    records = genotypes_to_records(relations, template, columns)
+
+    # Now we write the output of those to a file...
+    with tempfile.NamedTemporaryFile(mode='w', delete=False) as fd:
```

Comment:
`vcf.Writer` takes a stream, so you should be able to use `StringIO` and avoid the file system.